tools/telemetry-generator/test/memoryUsageTestHandler.test.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const handler = require("../src/handlers/memoryUsageTestHandler");
+
+describe("memoryUsageTestHandler", () => {
+	let mockLogger: any;
+	let mockFileData: any;
+
+	beforeEach(() => {
+		mockLogger = {
+			send: jest.fn(),
+		};
+
+		mockFileData = {
+			suiteName: "TestSuite",
+			benchmarks: [
+				{
+					benchmarkName: "Benchmark1",
+					customData: {
+						"Heap Used Avg": "123.45",
+						"Heap Used StdDev": "67.89",
+					},
+				},
+			],
+		};
+	});
+
+	it("should log endpoint name if defined", () => {
+		process.env.FLUID_ENDPOINTNAME = "testEndpoint";
+		const consoleLogSpy = jest.spyOn(console, "log").mockImplementation(() => {});
+
+		handler(mockFileData, mockLogger);
+
+		expect(consoleLogSpy).toHaveBeenCalledWith("ENDPOINTNAME", "testEndpoint");
+
+		consoleLogSpy.mockRestore();
+		delete process.env.FLUID_ENDPOINTNAME;
+	});
+
+	it("should log default endpoint name if not defined", () => {
+		const consoleLogSpy = jest.spyOn(console, "log").mockImplementation(() => {});
+
+		handler(mockFileData, mockLogger);
+
+		expect(consoleLogSpy).toHaveBeenCalledWith(
+			"ENDPOINTNAME not defined using local as default.",
+		);
+
+		consoleLogSpy.mockRestore();
+	});
+
+	it("should emit heap used avg and std dev metrics to logger", () => {
+		handler(mockFileData, mockLogger);
+
+		expect(mockLogger.send).toHaveBeenCalledTimes(1);
+
+		expect(mockLogger.send).toHaveBeenCalledWith({
+			namespace: "FFEngineering",
+			category: "performance",
+			eventName: "Benchmark",
+			benchmarkType: "MemoryUsage",
+			driverEndpointName: "",
+			suiteName: "TestSuite",
+			testName: "Benchmark1",
+			heapUsedAvg: "123.45",
+			heapUsedStdDev: "67.89",
+		});
+	});
+
+	it("should throw an error for invalid heap used avg metric", () => {
+		mockFileData.benchmarks[0].customData["Heap Used Avg"] = "invalid";
+
+		expect(() => handler(mockFileData, mockLogger)).toThrow(
+			"skipping metric Benchmark1 with value invalid as it is not a number",
+		);
+	});
```

Comment:
We should have tests like these for the execution time handler too, to validate the behavior when we receive invalid values.

Reply:
Added